include/kernel.h

```diff
@@ -3439,6 +3439,83 @@ __syscall int k_mutex_unlock(struct k_mutex *mutex);
  * @}
  */
 
+
+struct k_condvar {
+	_wait_q_t wait_q;
+};
+
+#define Z_CONDVAR_INITIALIZER(obj)                                             \
+	{                                                                      \
+		.wait_q = Z_WAIT_Q_INIT(&obj.wait_q),                          \
+	}
+
+/**
+ * @defgroup condvar_apis Condition Variables APIs
+ * @ingroup kernel_apis
+ * @{
+ */
+
+/**
+ * @brief Initialize a condition variable
+ *
+ * @param condvar pointer to a @p k_condvar structure
+ * @retval 0 Condition variable created successfully
+ */
+__syscall int k_condvar_init(struct k_condvar *condvar);
+
+/**
+ * @brief Signals one thread that is pending on the condition variable
+ *
+ * @param condvar pointer to a @p k_condvar structure
+ * @retval 0 On success
```

Comment:
Should we also document that all values !=0 are error, or just <0 is error? The documentation is ambiguous here

Reply:
I like negative is error, non-negative is success, but agree that potential error returns should be documented.  As is the natural question is "if it can only return 0, why isn't it `void`?"

Reply:
As a general note, I would argue that even if the function does not return <0, it might in the future if more error checks or conditions are added. In that case it would make sense to return int, even if we always return >= 0. It is difficult to change the API afterwards.